tests/libtest/CMakeLists.txt

```diff
@@ -16,6 +16,12 @@ function(setup_test TEST_NAME)          # ARGN are the files in the test
 
   target_link_libraries(${TEST_NAME} libcurl ${CURL_LIBS})
 
+  if(CURL_HAS_LTO)
+    set_target_properties(${TEST_NAME} PROPERTIES
+      INTERPROCEDURAL_OPTIMIZATION_RELEASE TRUE
+      INTERPROCEDURAL_OPTIMIZATION_RELWITHDEBINFO TRUE)
+  endif()
+
```

Comment:
Do you really want to do this for the test stuff? (anything under tests/) ? Optimizing them seems a bit unnecessary.

Reply:
Well. I added that first to make sure the optimization didn't break any test cases. Then I couldn't get the cmake build system to execute the tests at all. But you are right it probably shouldn't do it for production. I can change it.

Only libcurl and curl binary would be enough I think.